HW1/P4/P4.py

```diff
@@ -0,0 +1,54 @@
+
+
+import pickle
+import string
+import re
+import urllib2
+import csv
+import itertools
+import math as ma
+import os
+import P4_bfs
+
+
+if __name__ == '__main__':
+
+
+	partition = 16
+
+	output_rdd_captain_america = P4_bfs.bfs(start_node = 'CAPTAIN AMERICA', number_interations = 10, partition = partition)
+
+	node_distances_captain_america = output_rdd_captain_america.bfs_search()
+
+	print 'The max node distance for Captain America is', node_distances_captain_america[-1][1]
+
+	print 'The number of nodes in Captain Americas graph is', len(node_distances_captain_america)
+
+	pickle.dump(node_distances_captain_america, open( "P4_captain_america.pkl", "wb" ) )

```

Comment:
I wasn't familiar with the pickle module. It seems rather useful for cases such as these when working with objects and files.